src/bundle.c

```diff
@@ -347,15 +444,24 @@ gboolean check_bundle(const gchar *bundlename, gsize *size, gboolean verify, GEr
 	}
 
 	if (verify) {
+		CMS_ContentInfo *cms = NULL;
+		X509_STORE *store = NULL;
+
 		g_message("Verifying bundle... ");
 		/* the squashfs image size is in offset */
-		res = cms_verify_file(bundlename, sig, offset, &ierror);
+		res = cms_verify_file(bundlename, sig, offset, &cms, &store, &ierror);
 		if (!res) {
 			g_propagate_error(error, ierror);
 			goto out;
 		}
+
+		ibundle->signer = cms_get_signer_info(cms, store,
+				&ibundle->verified_chain);
 	}
 
+	if (bundle)
+		*bundle = ibundle;
```

Comment:
Shouldn't `ibundle` (and `ibundle->signer`) be free'd in case `bundle` evaluates to false?

Reply:
This is handled below in the updated version.